RockWeb/Blocks/Crm/PersonSearch.ascx.cs

```diff
@@ -262,6 +262,10 @@ private void BindGrid()
                         }
                 }
 
+                var personIdList = people.Select( p => p.Id ).ToList();
+
+                people = personService.Queryable(true).Where( p => personIdList.Contains( p.Id ) );
```

Comment:
this might not work when there are 1000s of personIds since it is going to turn into a SELECT * FROM PERSON WHERE ID IN (1,2,3,....) type of query.  You might want to test it in that scenario.